keras_cv/layers/preprocessing/cut_mix.py

```diff
@@ -0,0 +1,149 @@
+            input_shape[1],
+            input_shape[2],
+        )
+
+        lambda_sample = CutMix._sample_from_beta(
+            self.alpha, self.alpha, (tf.shape(labels)[0],)
+        )
+
+        ratio = tf.math.sqrt(1 - lambda_sample)
+
+        cut_height = tf.cast(
+            ratio * tf.cast(image_height, dtype=tf.float32), dtype=tf.int32
+        )
+        cut_width = tf.cast(
+            ratio * tf.cast(image_height, dtype=tf.float32), dtype=tf.int32
+        )
+
+        random_center_height = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_height, dtype=tf.int32
+        )
+        random_center_width = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_width, dtype=tf.int32
+        )
+
+        bbox_area = cut_height * cut_width
+        lambda_sample = 1.0 - bbox_area / (image_height * image_width)
+        lambda_sample = tf.cast(lambda_sample, dtype=tf.float32)
+
+        images = tf.map_fn(
+            lambda x: _fill_rectangle(*x),
+            (
+                images,
+                random_center_width,
+                random_center_height,
+                cut_width // 2,
+                cut_height // 2,
+                tf.reverse(images, [0]),
+            ),
+            dtype=(tf.float32, tf.int32, tf.int32, tf.int32, tf.int32, tf.float32),
+            fn_output_signature=tf.TensorSpec(images.shape[1:], dtype=tf.float32),
+        )
+
+        return images, labels, lambda_sample
+
+    def _update_labels(self, images, labels, lambda_sample):
+        labels_1 = self._smooth_labels(labels)
+        cutout_labels = tf.reverse(labels_1, [0])
+
+        lambda_sample = tf.reshape(lambda_sample, [-1, 1])
+        labels = lambda_sample * labels_1 + (1.0 - lambda_sample) * cutout_labels
+
+        return images, labels
+
+    def _smooth_labels(self, labels):
+        label_smoothing = self.label_smoothing or 0.0
+        off_value = label_smoothing / tf.cast(tf.shape(labels)[1], tf.float32)
+        on_value = 1.0 - label_smoothing + off_value
+        return labels * on_value + (1 - labels) * off_value
+
+
+def _fill_rectangle(
+    image, center_width, center_height, half_width, half_height, replace=None
+):
+    """Fill blank area."""
+    image_height = tf.shape(image)[0]
+    image_width = tf.shape(image)[1]
+
+    lower_pad = tf.maximum(0, center_height - half_height)
+    upper_pad = tf.maximum(0, image_height - center_height - half_height)
+    left_pad = tf.maximum(0, center_width - half_width)
+    right_pad = tf.maximum(0, image_width - center_width - half_width)
+
+    cutout_shape = [
+        image_height - (lower_pad + upper_pad),
+        image_width - (left_pad + right_pad),
+    ]
+    padding_dims = [[lower_pad, upper_pad], [left_pad, right_pad]]
+    mask = tf.pad(
+        tf.zeros(cutout_shape, dtype=image.dtype), padding_dims, constant_values=1
+    )
+    mask = tf.expand_dims(mask, -1)
+    mask = tf.tile(mask, [1, 1, 3])
```

Comment:
for the magic number 3 here, are we assuming the channel is always 3? should we use input_shape[-1]?